compiler/util/files.cpp

```diff
@@ -173,7 +173,12 @@ const char* makeTempDir(const char* dirPrefix) {
   removeSpacesBackslashesFromString(myuserid);
 
   const char* tmpDir = astr(tmpdirprefix, myuserid, tmpdirsuffix);
-  const char* dirRes = mkdtemp((char*) tmpDir);
+  char* tmpDirMut = strdup(tmpDir);  // LEAK, but necessary to modify an astr
```

Comment:
it won't leak if you do something like this at the end

```
const char* ret = astr(dirRes);
free(tmpDirMut);
return ret;
```

Reply:
Incorporated that change thanks